onedal/linear_model/tests/test_incremental_linear_regression.py

```diff
@@ -0,0 +1,151 @@
+    y_train_split = np.array_split(y_train, 2)
+    model = IncrementalLinearRegression(fit_intercept=True)
+    for i in range(2):
+        model.partial_fit(X_train_split[i], y_train_split[i], queue=queue)
+    model.finalize_fit()
+    y_pred = model.predict(X_test, queue=queue)
+    assert mean_squared_error(y_test, y_pred) < 2396
+
+
+@pytest.mark.parametrize("queue", get_queues())
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_pickle(queue, dtype):
+    # TODO Implement pickling for oneDAL entities
+    pass
+
+
+@pytest.mark.parametrize("queue", get_queues())
+@pytest.mark.parametrize("num_blocks", [1, 2, 10])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_full_results(queue, num_blocks, dtype):
+    seed = 42
+    num_features, num_targets = 19, 7
+    num_samples_train, num_samples_test = 3500, 1999
+
+    gen = np.random.default_rng(seed)
+    intercept = gen.random(size=num_targets, dtype=dtype)
+    coef = gen.random(size=(num_targets, num_features), dtype=dtype).T
+
+    X = gen.random(size=(num_samples_train, num_features), dtype=dtype)
+    y = X @ coef + intercept[np.newaxis, :]
+    X_split = np.array_split(X, num_blocks)
+    y_split = np.array_split(y, num_blocks)
+
+    model = IncrementalLinearRegression(fit_intercept=True)
+    for i in range(num_blocks):
+        model.partial_fit(X_split[i], y_split[i], queue=queue)
+    model.finalize_fit()
+
+    if queue.sycl_device.is_gpu:
+        tol = 5e-3 if model.coef_.dtype == np.float32 else 1e-5
+    else:
+        tol = 2e-3 if model.coef_.dtype == np.float32 else 1e-5
+    assert_allclose(coef, model.coef_.T, rtol=tol)
+
+    tol = 2e-3 if model.intercept_.dtype == np.float32 else 1e-5
+    assert_allclose(intercept, model.intercept_, rtol=tol)
+
+    Xt = gen.random(size=(num_samples_test, num_features), dtype=dtype)
+    gtr = Xt @ coef + intercept[np.newaxis, :]
+
+    res = model.predict(Xt, queue=queue)
+
+    tol = 2e-4 if res.dtype == np.float32 else 1e-7
+    assert_allclose(gtr, res, rtol=tol)
+
+
+@pytest.mark.parametrize("queue", get_queues())
+@pytest.mark.parametrize("num_blocks", [1, 2, 10])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_no_intercept_results(queue, num_blocks, dtype):
+    seed = 42
+    num_features, num_targets = 19, 7
+    num_samples_train, num_samples_test = 3500, 1999
+
+    gen = np.random.default_rng(seed)
+    coef = gen.random(size=(num_targets, num_features), dtype=dtype).T
+
+    X = gen.random(size=(num_samples_train, num_features), dtype=dtype)
+    y = X @ coef
+
+    X_split = np.array_split(X, num_blocks)
+    y_split = np.array_split(y, num_blocks)
+
+    model = IncrementalLinearRegression(fit_intercept=False)
+    for i in range(num_blocks):
+        model.partial_fit(X_split[i], y_split[i], queue=queue)
+    model.finalize_fit()
+
+    if queue.sycl_device.is_gpu:
+        tol = 3e-3 if model.coef_.dtype == np.float32 else 1e-7
```

Comment:
why is the error here so high for fp32 vs fp64? Thats 3.5 orders of magnitude...

Reply:
i just copied that from linear_regression_test in the main and i guess here the tolerance should be the same. i don't think resolving such issues is in the scope of this PR. i can add TODO comment if you want

Reply:
I agree. I would say a ticket in the backlog is better than a TODO in this circumstance. Just shoot me the ticket and I would call this good.

Reply:
Please remove TODO and ticket to the backlog